lib/core/test/unit/Cardano/Wallet/DBSpec.hs

```diff
@@ -641,5 +622,18 @@ dbPropertyTests dbLayer = do
             (checkCoverage . (prop_parallelPut putPrivateKey readPrivateKey
                 (length . lrp @Maybe)))
 
+-- | Clean a database by removing all wallets.
 cleanDB :: Monad m => DBLayer m s t -> m (DBLayer m s t)
 cleanDB db = listWallets db >>= mapM_ (runExceptT . removeWallet db) >> pure db
+
+-- | Provide a DBLayer to a Spec that requires it. The database is initialised
+-- once, and cleared with 'cleanDB' before each test.
+withDB :: IO (DBLayer IO s t) -> SpecWith (DBLayer IO s t) -> Spec
+withDB create = beforeAll create . beforeWith cleanDB
+
+instance Arbitrary (Wallet (SeqState DummyTarget) DummyTarget) where
+    shrink _ = []
+    arbitrary = initWallet <$> arbitrary
+
+instance Eq (SeqState DummyTarget) where
+    _ == _ = True
```

Comment:
What's the rationale for this instance? And what is the justification for it to be always true ?

Reply:
I believe (cannot say in the name of Rodney) it is because it is transparent in the context of tests we deliver here

Reply:
It feels somewhat wrong though. Why would any `SeqState` be equal? I wonder where this enters into play.

Reply:
yes, we need to add :